packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -407,16 +407,26 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
                   if (!loadingPrimary && searchPrimaryResults && searchPrimaryResults.length > 0) {
                     return (
                       <ul className={styles.diagnosisList}>
-                        {searchPrimaryResults.map((diagnosis, index) => (
-                          <li
-                            role="menuitem"
-                            className={styles.diagnosis}
-                            key={index}
-                            onClick={() => handleAddDiagnosis(diagnosis, 'primaryDiagnosisSearch')}
-                          >
-                            {diagnosis.display}
-                          </li>
-                        ))}
+                        {searchPrimaryResults.map((diagnosis, index) => {
+                          const isPrimaryDiagnosisSelected = selectedPrimaryDiagnoses.some(
+                            (selectedDiagnosis) => diagnosis.display === selectedDiagnosis.display,
```

Comment:
Change looks how it should be.
2 small nits:
1. Can we have a function that looks into both primary and secondary diagnoses? The same code is used in 2 places.
2. Please compare the diagnoses not by `display` but by `UUID`.

Thanks!

Reply:
Sure!